presto-main/src/main/java/com/facebook/presto/sql/analyzer/Analyzer.java

```diff
@@ -68,18 +68,24 @@ public Analysis analyze(Statement statement, boolean isDescribe)
         return analysis;
     }
 
-    static void verifyNoAggregatesOrWindowFunctions(Metadata metadata, Expression predicate, String clause)
+    static void verifyNoAggregatesWindowOrGroupingFunctions(Metadata metadata, Expression predicate, String clause)
```

Comment:
I think I'm going to leave this for FB to decide if they want to name it something more broad. A more generic name would probably be `enforceValidSqlConstructs()`.